internal/postrs/proof.go

```diff
@@ -183,6 +218,16 @@ func (v *Verifier) VerifyProof(
 		},
 	}
 
+	if opts.powCreatorId != nil {
+		logger.With().Info("Verifying with PoW creator ID", zap.String("id", hex.EncodeToString(opts.powCreatorId)))
+		minerIdSliceHdr := (*reflect.SliceHeader)(unsafe.Pointer(&opts.powCreatorId))
```

Comment:
Why here not also just `C.Bytes(opts.PowCraetorId)` and free it on the go side afterwards?

`len` is always guaranteed to be 32 already checked before passed into this function and `cap` should be irrelevant for rust.